testv2/e2e/scenario_migrate_csi_ccm.go

```diff
@@ -73,33 +73,27 @@ func (scenario *scenarioMigrateCSIAndCCM) Run(t *testing.T) {
 	k1Old := install.kubeone(t)
 	waitKubeOneNodesReady(t, k1Old)
 
+	client := dynamicClientRetriable(t, k1Old)
+	cpTests := newCloudProviderTests(client, scenario.infra.Provider())
+	defer cpTests.cleanUp(t)
```

Comment:
cpTests.runWithCleanup()

Reply:
This is intentional. I want to ensure that once the CCM/CSI migration is done, the pod with PVC is Running and LB is accessible. For that, I intentionally defer cleanup until the end.

Reply:
You could `t.Cleanup()` in that case. And `cpTests.run()` can do it for you ;)

Reply:
`t.Cleanup()` defers until the very end of the whole test!

Reply:
I want it to work in the following way:

* non-CCM/CSI migration tests: cleanup as soon as possible, i.e. before starting Sonobuoy tests (that's why `defer` is useful here)
* CCM/CSI migration tests: cleanup at the end, after making sure everything is running after the migration (note that we also don't have Sonobuoy tests for CCM/CSI migration, which is completely okay)

The first case is to save resources/money and leave space for other tests to run. Sonobuoy tests can take 10-20 minutes (or even longer if you run a bigger suite), so by cleaning up earlier, we let more tests run in parallel.